### PR TITLE
Fixes #2167 Unread indication doesn't work for guests

### LINKED ISSUE #2167: Unread indication doesn't work for guests
As soon as a guest visits one thread in a forum the whole forum is marked as read regardless of other unread threads.

### PR DESCRIPTION
**Pull request by Stefan-ST**
Fixes #2167

$fid can be a CSV list with multiple forums so ```forum_permissions($fid);``` was apparently wrong. This is unrelated to the bug but fixing wasn't too difficult.

### CODE REVIEW

File: inc/functions_indicators.php [hunk 1]
```diff
@@ -97,11 +127,11 @@ function fetch_unread_count($fid)
 			$count = 0;
 
 			// We've read at least some threads, are they here?
-			$query = $db->simple_select("threads", "lastpost, tid, fid", "visible=1 AND closed NOT LIKE 'moved|%' AND fid IN ({$fid}) AND lastpost > '{$cutoff}'{$onlyview}", array("limit" => 100));
+			$query = $db->simple_select("threads", "lastpost, tid, fid", "visible=1 AND closed NOT LIKE 'moved|%' AND {$where} AND lastpost > '{$cutoff}'", array("limit" => 100));
 
 			while($thread = $db->fetch_array($query))
 			{
-				if(isset($threadsread[$thread['tid']]) && $thread['lastpost'] > (int)$threadsread[$thread['tid']] && isset($forumsread[$thread['fid']]) && $thread['lastpost'] > (int)$forumsread[$thread['fid']])
+				if((!isset($threadsread[$thread['tid']]) || $thread['lastpost'] > (int)$threadsread[$thread['tid']]) && (!isset($forumsread[$thread['fid']]) || $thread['lastpost'] > (int)$forumsread[$thread['fid']]))
```
Comment (Stefan-ST) [inline inc/functions_indicators.php]: A thread without a threadsread cookie is unread not read...

**Comment by Cu8eR** [comment]
@Stefan-ST

**Comment by euantorano** [comment]
@Stefan-ST what kind of position is this fix in? Is it complete? Been open for quite a while now, would be nice to get it merged.

**Comment by Stefan-ST** [comment]
@euantorano The ticket says it's "fixed" since October. ;)

**Comment by euantorano** [comment]
Alright, so we can close this then?

**Comment by Stefan-ST** [comment]
If you ensure it works as expected, yes.

**Pull request merged**